src/sentry/integrations/discord/client.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from requests import PreparedRequest
+
+from sentry import options
+from sentry.services.hybrid_cloud.util import control_silo_function
+from sentry.shared_integrations.client.proxy import IntegrationProxyClient, infer_org_integration
+from sentry.shared_integrations.exceptions import IntegrationError
+from sentry.shared_integrations.exceptions.base import ApiError
+from sentry.utils.json import JSONData
+
+from .utils import logger
+
+
+class DiscordClient(IntegrationProxyClient):
+    integration_name: str = "discord"
+    base_url: str = "https://discord.com/api/v10"
+
+    # https://discord.com/developers/docs/resources/guild#get-guild
+    GET_GUILD_URL = "/guilds/%s"
```

Comment:
optional nit: This method of string formatting is not often used in Python3 since we have `.format()` now. You can use a method like [what we do for azure devops](https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/vsts/client.py) to name the params of the URL, or implement something like `_get_guild_url(guild_id: str) -> str`

Reply:
Thanks for mentioning! I hadn't see this formatting method before, but stuck with it based on other integrations. I'll make that change